packages/fuel-gauge/src/contract-factory.test.ts

```diff
@@ -249,21 +269,43 @@ describe('Contract Factory', () => {
     );
   });
 
-  it('should not deploy contracts greater than 100KB', async () => {
+  it('should not deploy contracts greater than MAX_CONTRACT_SIZE via deploy contract', async () => {
     using launched = await launchTestNode();
     const {
       wallets: [wallet],
     } = launched;
 
-    const largeByteCode = `0x${'00'.repeat(112400)}`;
-    const factory = new ContractFactory(largeByteCode, StorageTestContractAbi__factory.abi, wallet);
+    const factory = new ContractFactory(largeContractHex, LargeContractAbi__factory.abi, wallet);
 
     await expectToThrowFuelError(
-      async () => factory.deployContract(),
+      () => factory.deployContract(),
       new FuelError(
         ErrorCode.CONTRACT_SIZE_EXCEEDS_LIMIT,
         'Contract bytecode is too large. Max contract size is 100KB'
       )
     );
   });
+
+  it('should deploy contracts greater than MAX_CONTRACT_SIZE via a loader contract', async () => {
+    using launched = await launchTestNode({
+      providerOptions: {
+        cacheUtxo: -1,
+      },
+    });
+
+    const {
+      wallets: [wallet],
+    } = launched;
+    const factory = new ContractFactory(largeContractHex, LargeContractAbi__factory.abi, wallet);
+
+    const deploy = await factory.deployContractLoader<LargeContractAbi>();
```

Comment:
What is the reasoning for having two deployment methods based on the contract size? At first impression, it seems to me that this should be transparent to the user and that all of this should be under the `deploy` method: if the contract is less than 100KB, deploy as usual; if it's more, deploy via blobs.

Reply:
I had this thought also, however the experiences will be different and maybe that warrants a separate function / flow.

a) Contract < max size, calls `deployContract` and we submit a single create tx. Contract ID can be inferred directly from the bytecode that the user passed to it.

b) Contract > max size, calls `deployLoaderContract`, we submit multiple blob txs and then a create tx. Contract ID cannot be inferred until we have the blob IDs, and has to be via the loader bytecode rather than the bytecode the user passed.

Consider that a lot of our users will be using browser wallets that requires manual intervention to approve transactions, for a 2 mb contract under testnet conditions this could mean approving 7 transactions.

IMO the flow differences warrants a different functions. The RS team also went for a [different function](https://github.com/FuelLabs/fuels-rs/pull/1472), I'll find their reasoning also.

cc @FuelLabs/sdk-ts

Reply:
@nedsalk Yes, the original idea was to have it working transparently under the hood.

@danielbate I wonder if delegating the contract sizing to the user would be an improvement.

I like the idea of a smart[er] `deploy` method internally calling `deploySingle` or `deployMulti`.

If we want to give users the choice, we could also export all three.

Reply:
@segfault-magnet shares the sentiment that the flow change, and deployment of different bytecode that the user passes warrants a separate, more transparent flow.

He also mentions as contract size is chain configurable, users may see a difference in behaviour when testing locally / testnet vs mainnet. 

I'm going to keep the functions separate and exported seperately, but I'll introduce the _deployMulti_ function at the top of `deployContract` with a size check. And we can continue to discuss which way we'd like to go.

> @danielbate I wonder if delegating the contract sizing to the user would be an improvement.

I'll definitely implement something of this ilk.

Reply:
Apologies, I expressed myself poorly. 🙂 I meant the opposite—delegating the task of checking contract size and deciding on method A or B to the user doesn't seem like an improvement.

Although I understand the flow is different, I'm still on the original train of thought—about internalizing this if/else and guiding the user from there with clear errors in case they happen. Maybe I'm missing the actual downsides? 

Perhaps having all the options + plus a happy path with smart defaults could work well.

At the very least, we must provide utilities around contract size checks.

- `deploy` (if/else —> A|B happy path)
  - `deploySingle`
  - `deployMulti`
- `getContractSize(bytecode)`
- `getContractMaxSizeLimit()`

Reply:
Understood. Currently I've implemented `deploy` (A|B), `deployContract` (ensures no breaking change) and `deployContractAsBlobs`. The [docs](https://github.com/FuelLabs/fuels-ts/blob/db/feat/blob-contract-deploys/apps/docs/src/guide/contracts/deploying-contracts.md) should give better context now.